lib/matplotlib/tests/test_axes.py

```diff
@@ -3600,6 +3600,32 @@ def test_errorbar():
     ax.set_title("Simplest errorbars, 0.2 in x, 0.4 in y")
 
 
+@image_comparison(['mixed_errorbar_polar_caps'], extensions=['png'],
+                  remove_text=True)
+def test_mixed_errorbar_polar_caps():
+    fig = plt.figure()
+    ax = plt.subplot(111, projection='polar')
+    theta = np.linspace(0, np.pi, 3, False)
+    r = [1.0]*3
+    ax.errorbar(theta, r, xerr=0.35, yerr=0.2, fmt="o")
+    _long_errorbar_polar_caps(ax)
+    _asymmetric_polar_caps(ax)
+
+
+def _long_errorbar_polar_caps(ax):
+    theta = np.linspace(0, 2 * np.pi, 2, False) + np.pi/2
+    r = [1.8, 2.2]
+    ax.errorbar(theta, r, xerr=0.8 * np.pi, yerr=0.2, fmt="o")
+
+
+def _asymmetric_polar_caps(ax):
+    theta = np.linspace(0, np.pi, 3, False) + np.pi
+    r = [1.0]*3
+    xerr = [[.3, .3, .2], [.2, .3, .3]]
+    yerr = [[.35, .5, .5], [.5, .35, .5]]
+    ax.errorbar(theta, r, xerr=xerr, yerr=yerr, fmt="o")
```

Comment:
For symmetry and simplicity (less indirection) I'd put everything in one function:

```suggestion
    # symmetric errorbars
    theta = np.linspace(0, np.pi, 3, False)
    r = [1.0]*3
    ax.errorbar(theta, r, xerr=0.35, yerr=0.2, fmt="o")

    # long errorbars
    theta = np.linspace(0, 2 * np.pi, 2, False) + np.pi/2
    r = [1.8, 2.2]
    ax.errorbar(theta, r, xerr=0.8 * np.pi, yerr=0.2, fmt="o")

    # asymmetric errorbars
    theta = np.linspace(0, np.pi, 3, False) + np.pi
    r = [1.0]*3
    xerr = [[.3, .3, .2], [.2, .3, .3]]
    yerr = [[.35, .5, .5], [.5, .35, .5]]
    ax.errorbar(theta, r, xerr=xerr, yerr=yerr, fmt="o")
```

Additionally, I suggest not to position the bars on the grid. Pixel-wise interaction (antialiasing) between grid and errorbars could make the test less reproducible in the long run, and this is not something we want to test.
We can either

1. deactivate the grid (if that does not look too odd - the grid could be a visual help to verify for humans that the angles are right)
2. Slightly move the points in *r* and *theta*. - Note also that using explicit numbers may be simpler than `np.linspace`; e.g. for the long errorbar you could use `theta = [1.5, 3]`.

Reply:
Ok, I have manually created some positions, and I have put overlapping back into the game. I have followed your advice and I have tuned the exact positions to avoid grid.
I added a comment to the test briefly summarizing the test purpose and hinting for future reference about the grid avoiding and possibility to turn them off if we arrive to problems.